apps/state/test/state/stops_on_route_test.exs

```diff
@@ -363,6 +416,18 @@ defmodule State.StopsOnRouteTest do
       # know that there are only three items.
       assert [_, _, _] = merge_ids(stop_ids)
     end
+
+    test "can use an override to order individual stops" do
+      stop_ids =
+        Enum.shuffle([
```

Comment:
This will make the tests only fail intermittently if there's a problem.
Perhaps you should always do a merge in both directions, and check that both work or just do the one order that'd be broken now.

Reply:
This seems harder to understand:
```elixir
      overrides = [[2, 4, 3, 5]]

      for stop_ids <- [
            [
              [1, 2, 3, 5, 6],
              [1, 2, 4, 5, 6]
            ],
            [
              [1, 2, 4, 5, 6],
              [1, 2, 3, 5, 6]
            ],
          ] do
          assert [1, 2, 4, 3, 5, 6] == merge_ids(stop_ids, overrides)
      end
    end
```

Since `Enum.shuffle` is dependent on the seed, a failure that's caused by a change for this is easily reproducible. We already have some property tests which also depend on randomness, so it's already a possibility we live with.

Reply:
Having some intermittently failing tests doesn't make a new one any less painful. How about this?
```ex
overrides = [[2, 3, 4, 5]]
# try both permutations so we know the result order is due to the override and not the input order.
assert [1, 2, 3, 4, 5, 6] == merge_ids([
  [1, 2, 3, 5, 6],
  [1, 2, 4, 5, 6]
], overrides)
assert [1, 2, 3, 4, 5, 6] == merge_ids([
  [1, 2, 4, 5, 6],
  [1, 2, 3, 5, 6]
], overrides)
```
Even if it is more verbose, that's less bad than intermittent test failures.

Reply:
This is hard to understand, since it's not obvious what is the difference between the two lists when you have them twice. You can tell that it's harder to understand by the comment that's almost as long as the test :)

We already use randomness to find issues with the code (property tests, test suite order randomization, sorting tests), and as long as the failure is reproducible with the seed every time, it's not really an intermittent failure in the same way that say a race condition (affected by timing outside the control of the test) would be. Like many things in programming, it's a trade-off, and it's a trade-off that the API uses in other places to balance the amount of test code we need to write and the space covered by our tests.